include/wil/resource.h

```diff
@@ -5241,13 +5241,19 @@ namespace wil
     {
         unique_hglobal_locked() = delete;
 
-        unique_hglobal_locked(_In_ STGMEDIUM& medium) : unique_any<void*, decltype(&::GlobalUnlock), ::GlobalUnlock>(medium.hGlobal)
+        unique_hglobal_locked(HGLOBAL global) : unique_any<void*, decltype(&::GlobalUnlock), ::GlobalUnlock>(global)
         {
             // GlobalLock returns a pointer to the associated global memory block and that's what callers care about.
-            m_globalMemory = GlobalLock(medium.hGlobal);
+            m_globalMemory = GlobalLock(global);
         }
 
-        // In the future, we could easily add additional constructor overloads such as unique_hglobal_locked(HGLOBAL) to make consumption easier.
+        unique_hglobal_locked(unique_hglobal &global) : unique_hglobal_locked(global.get())
+        {
+        }
+
+        unique_hglobal_locked(_In_ STGMEDIUM &medium) : unique_hglobal_locked(medium.hGlobal)

```

Comment:
while you are here
- remove the `_In_` SAL 
- align the `&` with the type .

This probably should be `const STGMEDIUM& medium`, but that might break some use.

Reply:
I have removed _In_ SAL and aligned the & with the type.